MapboxNavigation/SimulatedRoute.swift

```diff
@@ -123,12 +123,12 @@ fileprivate extension Array where Element == SimulatedLocation {
             let timeToTravel: TimeInterval = distance / speed
             
             if timeToTravel > 1 {
-                let newCoordinate = coordinate(at: speed, fromStartOf: [location.coordinate, nextLocation.coordinate])
-                let newLocation = SimulatedLocation(coordinate: newCoordinate!,
+                guard let newCoordinate = coordinate(at: speed, fromStartOf: [location.coordinate, nextLocation.coordinate]) else { return processedLocations }
```

Comment:
Calling return in the `guard` statement here will break out of the for loop prematurely. Seems like it might be better to `continue` instead of returning early.